Tools/ViewsEstimators.py

```diff
@@ -6,118 +6,333 @@
-                 'XGBClassifier': XGBClassifier(n_estimators=100,learning_rate=0.05,n_jobs=-2),
-                 'HGBRegressor': HistGradientBoostingRegressor(max_iter=200),
-                 'HGBClassifier': HistGradientBoostingClassifier(max_iter=200),
+        estimators = {
+                    'linear': LinearRegression(),
+                    'logistic': LogisticRegression(solver='liblinear', random_state=random_state),
+
+                    'LGBMRegressor': LGBMRegressor(n_estimators=250, random_state=random_state),
+                    'LGBMClassifier': LGBMClassifier(n_estimators=250, random_state=random_state),
+
+                    'RFRegressor': RandomForestRegressor(n_estimators=250, random_state=random_state),
+                    'RFClassifier': RandomForestClassifier(n_estimators=250, random_state=random_state),
+
+                    'XGBRFRegressor': XGBRFRegressor(n_estimators=100, learning_rate=0.05, random_state=random_state),
+                    'XGBRFClassifier': XGBRFClassifier(n_estimators=100, learning_rate=0.05, random_state=random_state),
+
+                    'GBMRegressor': GradientBoostingRegressor(n_estimators=200, random_state=random_state),
+                    'GBMClassifier': GradientBoostingClassifier(n_estimators=200, random_state=random_state),
+
+                    'XGBRegressor': XGBRegressor(n_estimators=100, learning_rate=0.05, random_state=random_state),
+                    'XGBClassifier': XGBClassifier(n_estimators=100, learning_rate=0.05, random_state=random_state),
+                    
+                    'HGBRegressor': HistGradientBoostingRegressor(max_iter=200, random_state=random_state),
+                    'HGBClassifier': HistGradientBoostingClassifier(max_iter=200, random_state=random_state),
                 }
+        estimator = estimators.get(estimator_name)
 
-        return funcs[func_name]
+        return estimator
 
+    # Define the fit method for the class to train the model
     def fit(self,
             X: Union[np.ndarray, pd.DataFrame],
             y: Union[np.ndarray, pd.Series]):
-        X, y = check_X_y(X, y, dtype=None,
+
+        # Use sklearn fucntionality to check X and y for consistent length and enforce X to be 2D and y 1D. 
+        # By default, X is checked to be non-empty and containing only finite values.
+        # Standard input checks are also applied to y, such as checking that y
+        # does not have np.nan or np.inf targets.
+        X, y = check_X_y(X, 
+                         y, 
+                         dtype=None,
                          accept_sparse=False,
                          accept_large_sparse=False,
-                         force_all_finite='allow-nan')
+                         force_all_finite=True) #'allow-nan'
+        
+        # Set the number of features seen during fit
+        self.n_features_in_ = X.shape[1]
 
         if X.shape[1] < 2:
```

Comment:
Surely then it should be:

if X.shape[1] == 1:

Not important but still....

Reply:
Agree :) Fixed.